pkg/console/subresource/managedclusteraction/create_oauth_client.go

```diff
@@ -11,15 +11,18 @@ import (
 	// managedclusterviewv1beta1 "github.com/open-cluster-management/multicloud-operators-foundation/pkg/apis/action/v1beta1"
 )
 
-func DefaultCreateOAuthClientAction(cn string, sec string, redirects []string) *unstructured.Unstructured {
+func DefaultCreateOAuthClientAction(cn string, sec string, redirects []string) (*unstructured.Unstructured, error) {
 	mca := CreateOAuthClientStub(cn)
-	unstructured.SetNestedField(mca.Object, api.CreateOAuthClientManagedClusterActionName, "metadata", "name")
-	unstructured.SetNestedField(mca.Object, cn, "metadata", "namespace")
-	unstructured.SetNestedStringMap(mca.Object, util.LabelsForManagedClusterResources(cn), "metadata", "labels")
-	unstructured.SetNestedField(mca.Object, sec, "spec", "kube", "template", "secret")
-	unstructured.SetNestedField(mca.Object, api.ManagedClusterOAuthClientName, "spec", "kube", "template", "metadata", "name")
-	unstructured.SetNestedStringSlice(mca.Object, redirects, "spec", "kube", "template", "redirectURIs")
-	return mca
+	err := unstructured.SetNestedField(mca.Object, api.CreateOAuthClientManagedClusterActionName, "metadata", "name")
+	err = unstructured.SetNestedField(mca.Object, cn, "metadata", "namespace")
+	err = unstructured.SetNestedStringMap(mca.Object, util.LabelsForManagedClusterResources(cn), "metadata", "labels")
+	err = unstructured.SetNestedField(mca.Object, sec, "spec", "kube", "template", "secret")
+	err = unstructured.SetNestedField(mca.Object, api.ManagedClusterOAuthClientName, "spec", "kube", "template", "metadata", "name")
+	err = unstructured.SetNestedStringSlice(mca.Object, redirects, "spec", "kube", "template", "redirectURIs")
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I didn't see a need for an early return between each unstructured.SetNested... call, but  we can do that if anyone feels otherwise.

Reply:
nit:
yes it does not make sense to check for errors after each call, but this way we can loose an error if multiple errors occur. For this case we have `"k8s.io/apimachinery/pkg/util/errors"` package.
```go
	var errors []error

	errors = append(errors, unstructured.SetNestedField(mca.Object, api.CreateOAuthClientManagedClusterActionName, "metadata", "name"))
	errors = append(errors, unstructured.SetNestedField(mca.Object, cn, "metadata", "namespace"))
        ...
	if err != nil {
		return nil, utilerrors.NewAggregate(errors)
	}
```

Reply:
Wouldn't the `errors` array contain nil values if we do it this way?

Reply:
Nevermind, I see that the NewAggregate func handles that.